src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceOverrideService.kt

```diff
@@ -75,4 +81,65 @@ class LicenceOverrideService(
       ),
     )
   }
+
+  fun changeDates(licenceId: Long, request: OverrideLicenceDatesRequest) {
+    val licence = licenceRepository.findById(licenceId).orElseThrow { EntityNotFoundException("$licenceId") }
+
+    val username = SecurityContextHolder.getContext().authentication.name
+
+    log.info(
+      buildString {
+        append("Current licence dates - ID $licenceId ")
+        append("CRD ${licence.conditionalReleaseDate} ")
+        append("ARD ${licence.actualReleaseDate} ")
+        append("SSD ${licence.sentenceStartDate} ")
+        append("SED ${licence.sentenceEndDate} ")
+        append("LSD ${licence.licenceStartDate} ")
+        append("LED ${licence.licenceExpiryDate} ")
+        append("TUSSD ${licence.topupSupervisionStartDate} ")
+        append("TUSED ${licence.topupSupervisionExpiryDate}")
+      },
+    )
+
+    log.info(
+      buildString {
+        append("Updated dates - ID $licenceId ")
+        append("CRD ${request.conditionalReleaseDate} ")
+        append("ARD ${request.actualReleaseDate} ")
+        append("SSD ${request.sentenceStartDate} ")
+        append("SED ${request.sentenceEndDate} ")
+        append("LSD ${request.licenceStartDate} ")
+        append("LED ${request.licenceExpiryDate} ")
+        append("TUSSD ${request.topupSupervisionStartDate} ")
+        append("TUSED ${request.topupSupervisionExpiryDate}")
+      },
+    )
+
+    val updatedLicenceEntity = licence.copy(
+      conditionalReleaseDate = request.conditionalReleaseDate,
+      actualReleaseDate = request.actualReleaseDate,
+      sentenceStartDate = request.sentenceStartDate,
+      sentenceEndDate = request.sentenceEndDate,
+      licenceStartDate = request.licenceStartDate,
+      licenceExpiryDate = request.licenceExpiryDate,
+      topupSupervisionStartDate = request.topupSupervisionStartDate,
+      topupSupervisionExpiryDate = request.topupSupervisionExpiryDate,
+      dateLastUpdated = LocalDateTime.now(),
+      updatedByUsername = username,
+    )
+
+    licenceRepository.saveAndFlush(updatedLicenceEntity)
+
+    auditEventRepository.saveAndFlush(
+      AuditEvent(
+        licenceId = licence.id,
+        detail = "ID ${licence.id} type ${licence.typeCode} status ${licence.statusCode} version ${licence.version}",
+        eventTime = LocalDateTime.now(),
+        eventType = AuditEventType.USER_EVENT,
+        username = username,
+        fullName = username,
+        summary = "Sentence dates overridden for ${licence.forename} ${licence.surname}: ${request.reason}",
+      ),
+    )
```

Comment:
why are we not using the `UpdateSentenceDateService`?

Reply:
UpdateSentenceDateService can potentially change the licence status code depending on the dates you provide it and also send a notification in some conditions. We wanted to avoid any changes other updating the dates